### ISSUE TITLE
No stratified splitting for betti number evaluation

### ISSUE THREAD

**Issue by AdityaKulshrestha** [aidos-lab/mantra-benchmarks#18]
In the papers, authors have mentioned to use stratified splitting for all the tasks, whereas in the code they are not considering stratified splitting for betti numbers.

https://github.com/aidos-lab/mantra-benchmarks/blob/fea7a80fdf0622f3582328ae2a372511372624c3/code/experiments/generate_configs.py#L298

**Comment by danielbinschmid** [comment]
We make the exception for betti numbers because we deal with a multi-label classification task with very high sparsity across label values here. So, naive stratification cannot be applied and involved strategies are needed to implement a proper stratification strategy. We did not implement such. Future work may resort to techniques like Iterative Stratification [1] to implement stratification for the betti number case, if that is what you are seeking for!


[1] Sechidis, Konstantinos, Grigorios Tsoumakas, and Ioannis Vlahavas. "On the stratification of multi-label data." Machine Learning and Knowledge Discovery in Databases: European Conference, ECML PKDD 2011, Athens, Greece, September 5-9, 2011, Proceedings, Part III 22. Springer Berlin Heidelberg, 2011.

**Comment by prinshul** [comment]
Hi,

The paper says that betti numbers are predicted with MSE loss and then values are rounded to the nearest integer.

"For Betti number prediction, we approached training as a multivariate regression task, since Betti numbers can theoretically be arbitrarily large. Our loss function in this case was the mean squared error, and the Betti number prediction was obtained by rounding the model outputs to the nearest integer."

**Comment by danielbinschmid** [comment]
Yes, this is correct.

**Comment by danielbinschmid** [comment]
However, the training loss is not related to stratification, which is the topic of this issue. May I ask why you mentioned this matter?

**Comment by prinshul** [comment]
> However, the training loss is not related to stratification, which is the topic of this issue. May I ask why you mentioned this matter?

Ideally, for multi-label classification tasks, we should use multi-class cross entropy instead of MSE. 

Referring to stratification, with a simple stratified approach, the variance is low in the betti number prediction task as well and doesn't seem to give results with high variance across the runs. Though I can check without it as well.

**Comment by danielbinschmid** [comment]
You are correct with the former, if we stick to multi-label classification. However, for betti numbers, out-of-domain test cases may happen with betti numbers out of bounds of the training dataset. In such case, the model will not be able to predict betti numbers which were not considered for training. The model should be able to predict such betti numbers.

Interesting finding about the simple stratification on the latter! This is in-line with the orientability task, recall that orientability corresponds to predicting the "last" betti number, i.e. b2 for 2-manifolds. We did not have high variance for predicting orientability too.
